File: server/router/chat_monitor/chat_monitor.go
package chat_monitor

import (
	"context"
	"sync"

	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/flipped-aurora/gin-vue-admin/server/router/chat_monitor/initialize"
	"github.com/gin-gonic/gin"
)

type RouterGroup struct {
	ChatMonitorRouter
}

type ChatMonitorRouter struct{}

var once sync.Once

func (r *ChatMonitorRouter) InitChatMonitorRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	// 注册菜单和API（只执行一次）
	once.Do(func() {
		ctx := context.Background()
		initialize.Menu(ctx)
		initialize.Api(ctx)
	})

	chatMonitorApi := v1.ApiGroupApp.ChatMonitorApiGroup

	// 需要操作记录的路由
	recordRouter := Router.Group("chatMonitor").Use(middleware.OperationRecord())
	{
		recordRouter.POST("game/create", chatMonitorApi.GameApi.CreateGame)
		recordRouter.PUT("game/update", chatMonitorApi.GameApi.UpdateGame)
		recordRouter.DELETE("game/delete", chatMonitorApi.GameApi.DeleteGame)

		recordRouter.POST("sensitive/create", chatMonitorApi.SensitiveApi.CreateSensitiveWord)
		recordRouter.POST("sensitive/import", chatMonitorApi.SensitiveApi.ImportSensitiveWords)
		recordRouter.PUT("sensitive/update", chatMonitorApi.SensitiveApi.UpdateSensitiveWord)
		recordRouter.DELETE("sensitive/delete", chatMonitorApi.SensitiveApi.DeleteSensitiveWord)

		recordRouter.POST("whitelist/create", chatMonitorApi.WhitelistApi.CreateWhitelist)

		recordRouter.POST("ban/create", chatMonitorApi.BanApi.CreateBan)
		recordRouter.PUT("ban/revoke", chatMonitorApi.BanApi.RevokeBan)
	}

	// 只需要鉴权的路由
	privateRouter := Router.Group("chatMonitor")
	{
		privateRouter.GET("game/list", chatMonitorApi.GameApi.GetGameList)

		privateRouter.GET("chat/history", chatMonitorApi.ChatApi.GetChatHistory)

		privateRouter.GET("sensitive/list", chatMonitorApi.SensitiveApi.GetSensitiveWordList)
		privateRouter.GET("whitelist/list", chatMonitorApi.WhitelistApi.GetWhitelistList)

		privateRouter.GET("ban/list", chatMonitorApi.BanApi.GetBanList)

		privateRouter.GET("stats/overview", chatMonitorApi.StatsApi.GetStatsOverview)
		privateRouter.GET("stats/trend", chatMonitorApi.StatsApi.GetStatsTrend)
		privateRouter.GET("stats/violators", chatMonitorApi.StatsApi.GetViolators)
	}
}

File: web/src/api/chatMonitor.js
import service from '@/utils/request'

// ========== 游戏管理 ==========
export const getGameList = (data) => {
  return service({ url: '/chatMonitor/game/list', method: 'get', params: data })
}
export const createGame = (data) => {
  return service({ url: '/chatMonitor/game/create', method: 'post', data })
}
export const updateGame = (data) => {
  return service({ url: '/chatMonitor/game/update', method: 'put', data })
}
export const deleteGame = (data) => {
  return service({ url: '/chatMonitor/game/delete', method: 'delete', data })
}

// ========== 聊天记录 ==========
export const getChatHistory = (data) => {
  return service({ url: '/chatMonitor/chat/history', method: 'get', params: data })
}

// ========== 敏感词 ==========
export const getSensitiveList = (data) => {
  return service({ url: '/chatMonitor/sensitive/list', method: 'get', params: data })
}
export const createSensitive = (data) => {
  return service({ url: '/chatMonitor/sensitive/create', method: 'post', data })
}
export const importSensitive = (data) => {
  return service({ url: '/chatMonitor/sensitive/import', method: 'post', data })
}
export const updateSensitive = (data) => {
  return service({ url: '/chatMonitor/sensitive/update', method: 'put', data })
}
export const deleteSensitive = (data) => {
  return service({ url: '/chatMonitor/sensitive/delete', method: 'delete', data })
}

// ========== 白名单 ==========
export const getWhitelistList = (data) => {
  return service({ url: '/chatMonitor/whitelist/list', method: 'get', params: data })
}
export const createWhitelist = (data) => {
  return service({ url: '/chatMonitor/whitelist/create', method: 'post', data })
}

// ========== 封禁管理 ==========
export const createBan = (data) => {
  return service({ url: '/chatMonitor/ban/create', method: 'post', data })
}
export const revokeBan = (data) => {
  return service({ url: '/chatMonitor/ban/revoke', method: 'put', data })
}
export const getBanList = (data) => {
  return service({ url: '/chatMonitor/ban/list', method: 'get', params: data })
}

// ========== 数据报表 ==========
export const getStatsOverview = (data) => {
  return service({ url: '/chatMonitor/stats/overview', method: 'get', params: data })
}
export const getStatsTrend = (data) => {
  return service({ url: '/chatMonitor/stats/trend', method: 'get', params: data })
}
export const getViolators = (data) => {
  return service({ url: '/chatMonitor/stats/violators', method: 'get', params: data })
}
